app/controllers/home_controller.rb

```diff
@@ -1,2 +1,33 @@
 class HomeController < ApplicationController
+  def index 
+    @products = ProductFilterService.new(params).filter
+
+    @products = filter_by_category(@products, params[:category])
+    @products = filter_by_vegetarian(@products, params[:vegetarian])
+
+    @selected_category = params[:category] || "Category"
```

Comment:
I would rather check if the param is present, if it is then filter it
```
@products = @products.where(category: params[:category]) if params[:category]
```
This way you don't need the extra method